lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc

```diff
@@ -123,25 +121,26 @@ function dosomething_reportback_get_reportbacks_query($params) {
   return $result;
 }
 
-
 /**
- * Filter out requested flagged Reportbacks based on user permissions.
- *
- * @param  array  $params
- * @return mixed
+ * Determine if user has access to flagged Reportbacks based on user permission. 
+ * @param array $results
  */
-function dosomething_reportback_filter_flagged_reportbacks($params) {
+function dosomething_reportback_accessible_results ($results) {
   if (user_access('view any reportback')) {
-    if (isset($params['flagged'])) {
-      unset($params['flagged']);
+    return $results;
+  }
+  else {
+    foreach ($results as $key => $reportback) {
+      if ($reportback->flagged) {
```

Comment:
Confirmed - with a reportback that is not flagged (`reportback->flagged = '0'`), `0` is a string. It was also returned to the non-admin user as expected (instead of hitting `403` status code with error message `Access denied.`).